upgrade/satellite.py

```diff
@@ -74,189 +71,150 @@ def satellite6_setup(os_version):
     return sat_host
 
 
-def satellite6_upgrade():
-    """Upgrades satellite from old version to latest version.
+def satellite6_upgrade(upgrade_type=None):
+    """This function is used to perform the satellite upgrade of two type
+        - Upgrades Satellite Server from old version to latest
+            The following environment variables affect this command:
 
-    The following environment variables affect this command:
+            BASE_URL
+                Optional, defaults to available satellite version in CDN.
+                URL for the compose repository
+            TO_VERSION
+                Satellite version to upgrade to and enable repos while upgrading.
+                e.g '6.1','6.2', '6.3'
+            PERFORM_FOREMAN_MAINTAIN_UPGRADE
+                use foreman-maintain for satellite upgrade
+
+        - Upgrades Satellite Server to its latest zStream version
+            Note: For zstream upgrade both 'To' and 'From' version should be same
+
+            FROM_VERSION
+                Current satellite version which will be upgraded to latest version
+            TO_VERSION
+                Next satellite version to which satellite will be upgraded
+            PERFORM_FOREMAN_MAINTAIN_UPGRADE
+                use foreman-maintain for satellite upgrade
 
-    BASE_URL
-        Optional, defaults to available satellite version in CDN.
-        URL for the compose repository
-    TO_VERSION
-        Satellite version to upgrade to and enable repos while upgrading.
-        e.g '6.1','6.2', '6.3'
-    PERFORM_FOREMAN_MAINTAIN_UPGRADE
-        use foreman-maintain for satellite upgrade
     """
     logger.highlight('\n========== SATELLITE UPGRADE =================\n')
     to_version = os.environ.get('TO_VERSION')
+    from_version = os.environ.get('FROM_VERSION')
+    if upgrade_type:
+        if not from_version == to_version:
```

Comment:
Upgrade will always fail for old to new upgrade as the versions will be different always.

Reply:
Correct, this part will work only when upgrade_type is zStream.